src/main/java/com/karankumar/bookproject/backend/service/PredefinedShelfService.java

```diff
@@ -44,14 +46,16 @@
     private final PredefinedShelfRepository predefinedShelfRepository;
     private final AuthorRepository authorRepository;
     private final TagRepository tagRepository;
+    private final UserService userService;
 
     public PredefinedShelfService(BookRepository bookRepository, AuthorRepository authorRepository,
-            PredefinedShelfRepository shelfRepository, TagRepository tagRepository) {
+            PredefinedShelfRepository shelfRepository, TagRepository tagRepository, UserService userService) {
```

Comment:
Do we really need `UserService` dependency in `ShelfService`? All we need from `UserService` is the current user so wouldn't it be better to have this at the repo layer? There may not be any change at the service layer and we pass the current logged in user to shelf repo?

Reply:
I don't understand what you propose...

Reply:
I was thinking about if the repository layer would be more appropriate for this change or service layer. But I think it's fine in service layer as well.

Reply:
@Burhak For consistency, could you please pass in the `UserRepository` here?